### PR TITLE
Add ERC: ERC-721 Name Registry Extension

### PR DESCRIPTION
**Pull request by chenly**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: ERCS/erc-9999.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
+---
+eip: 9999
```
Comment (abcoathup) [inline ERCS/erc-9999.md:2]:
```suggestion
eip: 7644
```
Assigning next sequential EIP/ERC/RIP number.
Numbering changed to sequential from 7500.

Please also update the filename.

File: ERCS/erc-9999.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+title: ERC-721 Name Registry Extension
+description: Add a Time-Limited unique naming registry mechanism to ERC-721 tokens. 
+author: Chen Liaoyuan (@chenly)
+discussions-to: https://ethereum-magicians.org/t/erc-9999-erc-721-name-registry-extension/19005
```
Comment (abcoathup) [inline ERCS/erc-9999.md:6]:
```suggestion
discussions-to: [Please add link to topic in https://ethereum-magicians.org]
```

File: ERCS/erc-7644.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
+---
+eip: 7644
+title: ERC-721 Name Registry Extension
+description: Add a time-limited unique naming registry mechanism to ERC-721 tokens.
```
Comment (SamWilsn) [inline ERCS/erc-7644.md:4]: Decent description&mdash;it does introduce more information than the title&mdash;but my concern here is that it doesn't really explain what a "naming registry" actually is.

File: ERCS/erc-7644.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-03-01
+requires: 721
+---
+
+## Abstract
+
+This extension defines an interface that adds a naming mechanism to [ERC-721](./eip-721.md) tokens. It allows each token to have a unique name with a set expiration date. The interface includes functions for assigning, updating, and querying names and their associated tokens, ensuring that names remain unique until they expire.
```
Comment (SamWilsn) [inline ERCS/erc-7644.md:16]: I'd like to see some text explaining the scope and purpose of the names. For example:

- are the names globally unique or just for the specific token contract?
- who sets the names, and who is expected to use the names?

File: ERCS/erc-7644.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+pragma solidity ^0.8.0;
+
+/**
+ * @title INameRegistry
+ * @dev Interface for the NameRegistry smart contract.
+ * This interface allows interaction with a NameRegistry, 
+ * enabling the registration, management, and lookup of names 
+ * with associated expiry dates tied to specific tokens.
+ */
+interface INameRegistry {
```
Comment (SamWilsn) [inline ERCS/erc-7644.md:34]: It's unclear if this is supposed to be implemented by the same contract implementing `ERC721` or if this is a completely separate contract.

File: ERCS/erc-7644.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+     */
+    function getName(bytes32 _name) external view returns (uint256 tokenId, uint256 expiryDate);
+	
+}
+```
+
+## Rationale
+
+Here are a few design decisions and why they were made:
+
```
Comment (SamWilsn) [inline ERCS/erc-7644.md:98]:
```suggestion
```

File: ERCS/erc-7644.md [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+#### Username Restrictions
+
+To facilitate indexing and gas efficiency, usernames should adhere to a length constraint of 3 to 32 characters. This range prevents the registration of overly long names, which can be costly in terms of gas and difficult to manage. Limiting characters to the range of [a-zA-Z0-9] enhances readability and prevents the abuse of the naming system by restricting the use of special characters that could complicate domain resolution or user recognition. Implementing these constraints not only promotes a high level of usability within the ecosystem but also guards against the proliferation of spam registrations, ensuring that the registry remains accessible and functional for genuine users.
```
Comment (SamWilsn) [inline ERCS/erc-7644.md:193]: My understanding after reading this is that these are optional suggestions that implementers should be aware of. That seems reasonable to me, but if you want these to be _mandatory_ restrictions, you should mention them in the Specification section.

**Review by SamWilsn — APPROVED** [review]

**Pull request merged**